src/crypto-api.ts

```diff
@@ -591,6 +591,20 @@ export interface ImportRoomKeyProgressData {
     total: number;
 }
 
+/**
+ * Where a room key came from
+ */
+export enum RoomKeySource {
+    /** from key backup */
+    Backup = "backup",
+    /** from a key export */
+    Export = "export",
+    /** via a key forward */
+    Forward = "forward",
+    /** directly from the sender */
+    Direct = "direct",
```

Comment:
Are `Forward` and `Direct` ever actually used? It seems like they are not, and since such cases are handled internally within `matrix-sdk-crypto`, it seems like they never actually will be? I don't think we should have options which will never actually be used.